### PR TITLE
Add StakingFeature

### LINKED ISSUE #890: Add StakingFeature - rust
https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#staking-feature

### PR DESCRIPTION
**Pull request by thibault-martinez**
oops

### CODE REVIEW

File: sdk/src/types/block/output/feature/staking.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.end_epoch
+    }
+}
+
+#[allow(missing_docs)]
+pub mod dto {
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub struct StakingFeatureDto {
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/feature/staking.rs:60]: serde camelCase?

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/staking.rs:60]: Done

File: sdk/src/types/block/output/feature/staking.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct StakingFeatureDto {
+        #[serde(rename = "type")]
+        pub kind: u8,
+        pub staked_amount: u64,
+        pub fixed_cost: u64,
+        pub start_epoch: u64,
+        pub end_epoch: u64,
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/feature/staking.rs]: Actually since these are u64 they should  be serialized as String

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/staking.rs]: They are all integers in the API TIP

Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/staking.rs]: Yeah but that's the packed serialization right?

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/staking.rs]: No, they are also u64 in the JSON HTTP API

Comment (Thoralf-M) [reply sdk/src/types/block/output/feature/staking.rs]: Asked there https://github.com/iotaledger/tips-draft/pull/24#discussion_r1276251914

Comment (DaughterOfMars) [reply sdk/src/types/block/output/feature/staking.rs]: Well, that's a problem because those will not do the round trip properly.

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/staking.rs]: Changed

File: sdk/src/types/block/output/feature/staking.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+pub(crate) mod dto {
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct StakingFeatureDto {
+        #[serde(rename = "type")]
+        pub kind: u8,
+        #[serde(with = "crate::utils::serde::string")]
```
Comment (DaughterOfMars) [inline sdk/src/types/block/output/feature/staking.rs]: One import?

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/staking.rs]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**